src/PKSim.Core/Services/MoleculeExpressionTask.cs

```diff
@@ -67,7 +71,11 @@ public ICommand RemoveExpressionProfileFor(IndividualMolecule molecule, TSimulat
       public ICommand AddExpressionProfile(TSimulationSubject simulationSubject, ExpressionProfile expressionProfile)
       {
          var moleculeFactory = _individualMoleculeFactoryResolver.FactoryFor(expressionProfile.Molecule);
+         //this calls not only creates a new molecule but modify the subject in place to have the link to the expression profile
+         //by adding sub containers in the individual structure. The factory does not register the container and parameters so we need to do it
          var molecule = moleculeFactory.AddMoleculeTo(simulationSubject, expressionProfile.MoleculeName);
+         //we register the whole individual again.
+         _registrationTask.Register(simulationSubject);
```

Comment:
This was the problem here.
The parameters of the molecule in the individual were not registered (only at creation time...when loading the project, they were registered=>NO BUG)

Reply:
there is really not much else to this bug.. But this took me one day lol

Reply:
Cool, well done! 👍

Reply:
Actually this is the right fix but not the right place. I need to move this to the command (symmetrical to the unregister)